### PR TITLE
[OSF-7601] [OSF-7609] Profile page optimization and fixes

### PR DESCRIPTION
**Pull request by sloria**
## Purpose

The profile page currently loads very slowly. The root cause of this is the number of "embedded" requests that `mod-meta` makes when rendering the page. These embedded requests result in a lot of redundant querying.

This removes the embedded requests by nesting the node list data in the Mako-rendered views and passing the data to Mako defs (thus removing mod-meta).

## Changes

- Endpoints for `/profile/`, `/<nid>/`, `/<nid>/registrations/`, `/<nid>/forks/` return nested data in their 
- `render_nodes.mako` and `render_node.mako` were transformed into mako `defs`.
- Unused APIv1 endpoints were removed
- Serialization helpers were moved from `website.project.views.node` to `website.views` to prevent circular imports

## Side effects

- Profile page, project page, registrations list, and forks list should all load faster
- Removes APIv1 endpoints that are no longer used

## Tickets

- https://openscience.atlassian.net/browse/OSF-7601
- fixes https://openscience.atlassian.net/browse/OSF-7612
- fixes https://openscience.atlassian.net/browse/OSF-7609

### CODE REVIEW

**Comment by sloria** [comment]
TCI passed on my fork (up-to-date with `develop` as of ~1 hr ago).

File: tests/test_serializers.py [hunk 1]
```diff
@@ -255,6 +231,204 @@ def test_view_project_pending_registration_for_write_contributor_does_not_contai
...
+
+    def test_view_project_embed_descendants(self):
+        child = NodeFactory(parent=self.project, creator=self.user)
+        res = _view_project(self.project, auth=Auth(self.project.creator), embed_descendants=True)
+        assert_in('descendants', res['node'])
+        assert_equal(len(res['node']['descendants']), 1)
+        assert_equal(res['node']['descendants'][0]['id'], child._id)
+
+
+class TestGetReadableDescendants(OsfTestCase):
```
Comment (sloria) [inline tests/test_serializers.py]: These used to be view tests. I removed `/api/v1/<nid>/get_readable_descendants/`, so I ported the tests into serializer tests.

File: website/templates/util/contributor_list.mako [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<%def name="render_contributor_dict(contributor)">
```
Comment (sloria) [inline website/templates/util/contributor_list.mako]: These were adapted from `render_contributors_abbrev.mako`.

**Comment by icereval** [comment]
I'm seeing the same as @mfraezz, load times do not appear to be different at all on my build, maybe I'm doing something wrong, lets discuss.

**Comment by sloria** [comment]
Hm, I'll take a look at the load times again when I'm done with the migration I'm working on.

**Pull request merged**